util.js

```diff
@@ -236,13 +236,20 @@
         customFavicon = domain.match(faviconDomains);
 
         if (customFavicon) {
-            // check for dark background & available variant
-            if (isDarkBg && darkVariants.test(customFavicon)) {
-                variant = '.white';
+            // validate domain
+            customFavicon = customFavicon.toString();
+            tokens = sourceUrl.split('.');
+            domainMatch = (tokens.length > 1) ? (tokens[tokens.length - 2] === customFavicon) : true;
+          
+            if (domainMatch) {
+                // check for dark background & available variant
+                if (DDG.settings && DDG.settings.get('kae') === 'd' && darkVariants.test(customFavicon)) {
```

Comment:
The logic to check for dark theme was off. I also moved the check to be inside the IF condition i.e. there's no point checking the value unless it is for a customFavicon that matches.